apps/glusterfs/app_db.go

```diff
@@ -188,3 +193,184 @@ func (a *App) DbDump(w http.ResponseWriter, r *http.Request) {
 		panic(err)
 	}
 }
+
+// DbCreate ... Creates a bolt db file based on JSON input
+func DbCreate(jsonfile string, dbfile string) error {
+	var dump Db
+	//vars := mux.Vars(r)
+	//jsonFile := vars["jsonFile"]
+	// Check arguments
+	//if jsonFile == "" {
+	//	logger.Info("rtalurlogs, jsonFile value is %v", jsonFile)
+	//	http.Error(w, ErrNotFound.Error(), http.StatusInternalServerError)
+	//	return
+	//}
+
+	// Load config file
+	fp, err := os.Open(jsonfile)
+	if err != nil {
+		return fmt.Errorf("Could not open input file: %v", err.Error())
+	}
+	defer fp.Close()
+
+	dbParser := json.NewDecoder(fp)
+	if err = dbParser.Decode(&dump); err != nil {
+		return fmt.Errorf("Could not decode input file as JSON: %v", err.Error())
+	}
+
+	// Setup BoltDB database
+	dbhandle, err := bolt.Open(dbfile, 0600, &bolt.Options{Timeout: 3 * time.Second})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to open database: %v. exiting", err)
+		os.Exit(1)
+	} else {
+		err = dbhandle.Update(func(tx *bolt.Tx) error {
+			// Create Cluster Bucket
+			_, err := tx.CreateBucketIfNotExists([]byte(BOLTDB_BUCKET_CLUSTER))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Unable to create cluster bucket in DB")
+				return err
+			}
+
+			// Create Node Bucket
+			_, err = tx.CreateBucketIfNotExists([]byte(BOLTDB_BUCKET_NODE))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Unable to create node bucket in DB")
+				return err
+			}
+
+			// Create Volume Bucket
+			_, err = tx.CreateBucketIfNotExists([]byte(BOLTDB_BUCKET_VOLUME))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Unable to create volume bucket in DB")
+				return err
+			}
+
+			// Create Device Bucket
+			_, err = tx.CreateBucketIfNotExists([]byte(BOLTDB_BUCKET_DEVICE))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Unable to create device bucket in DB")
+				return err
+			}
+
+			// Create Brick Bucket
+			_, err = tx.CreateBucketIfNotExists([]byte(BOLTDB_BUCKET_BRICK))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Unable to create brick bucket in DB")
+				return err
+			}
+
+			_, err = tx.CreateBucketIfNotExists([]byte(BOLTDB_BUCKET_BLOCKVOLUME))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Unable to create blockvolume bucket in DB")
+				return err
+			}
+
+			_, err = tx.CreateBucketIfNotExists([]byte(BOLTDB_BUCKET_DBATTRIBUTE))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Unable to create dbattribute bucket in DB")
+				return err
+			}
+
+			return nil
+
+		})
+		if err != nil {
+			logger.Err(err)
+			return nil
+		}
+	}
+
+	//this registration should ideally be done during initialization, but it is existing bug
```

Comment:
In the comment could you elaborate a bit more what this means? As a newcomer to the code I don't have the history to know that existing bug this could be referring to.

Reply:
done